src/js/app.js

```diff
@@ -57,6 +59,31 @@ var app = {
         $('#modalDelete-link').attr('href', $(this).data('link'));
         $('#modalDelete').modal('show');
     },
+    modalVanity: function () {
+        var id = $(this).data('id');
+        $('#modalVanity').modal('show');
+        $('#modalVanity-link').click(function () {
+            var $callerButton = $(this);
+            $.post(window.AppConfig.base_url + '/upload/' + id + '/vanity', {vanity: $('#modalVanity-input').val()}, function (data, status) {
+                $callerButton.tooltip('dispose');
+                $('#modalVanity').modal('hide');
+                $('#modalVanity-input').val('');
+                var data = JSON.parse(data);
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏data is redeclared, please use npm to install the dependencies and run npx grunt to execute the js compilation, it will highlight missing semicolons, redeclarations, etc.